src/descriptor.rs

```diff
@@ -40,6 +40,7 @@ pub struct TerminalPath {
 }
 
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+#[derive(Serialize, Deserialize)]
```

Comment:
This doesn't need serde since has to be serialized as string representation. If you need binary serialziation as well, than we need a custom serde implementation which serializes as string for `human_readable` case